test/integration/extensions_test.py

```diff
@@ -0,0 +1,184 @@
+        expected = {
+            'user_id': {'type': 'DECIMAL', 'precision': 18, 'scale': 0},
+            'user_name': {'type': 'VARCHAR', 'size': 255, 'characterSet': 'UTF8'},
+            'register_dt': {'type': 'DATE', 'size': 4},
+            'last_visit_ts': {'type': 'TIMESTAMP', 'withLocalTimeZone': False, 'size': 8},
+            'is_female': {'type': 'BOOLEAN'},
+            'user_rating': {'type': 'DECIMAL', 'precision': 10, 'scale': 5},
+            'user_score': {'type': 'DOUBLE'},
+            'status': {'type': 'VARCHAR', 'size': 50, 'characterSet': 'UTF8'}
+        }
+        actual = connection.ext.get_columns_sql('SELECT * FROM users')
+        assert expected == actual
+
+    def test_get_sys_columns(self, connection):
+        expected = [
+            {'name': 'USER_ID', 'type': 'DECIMAL', 'sql_type': 'DECIMAL(18,0)', 'size': 18, 'scale': 0, 'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'USER_NAME', 'type': 'VARCHAR', 'sql_type': 'VARCHAR(255) UTF8', 'size': 255, 'scale': None,
+             'nulls': True, 'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'REGISTER_DT', 'type': 'DATE', 'sql_type': 'DATE', 'size': 10, 'scale': None, 'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'LAST_VISIT_TS', 'type': 'TIMESTAMP', 'sql_type': 'TIMESTAMP', 'size': 29, 'scale': None,
+             'nulls': True, 'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'IS_FEMALE', 'type': 'BOOLEAN', 'sql_type': 'BOOLEAN', 'size': 1, 'scale': None, 'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'USER_RATING', 'type': 'DECIMAL', 'sql_type': 'DECIMAL(10,5)', 'size': 10, 'scale': 5,
+             'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'USER_SCORE', 'type': 'DOUBLE PRECISION', 'sql_type': 'DOUBLE', 'size': 64, 'scale': None,
+             'nulls': True, 'distribution_key': False, 'default': None, 'comment': None},
+            {'name': 'STATUS', 'type': 'VARCHAR', 'sql_type': 'VARCHAR(50) UTF8', 'size': 50, 'scale': None,
+             'nulls': True,
+             'distribution_key': False, 'default': None, 'comment': None}
+        ]
+        actual = connection.ext.get_sys_columns('users')
+        assert expected == actual
+
+    def test_get_sys_tables(self, connection):
+        expected = [
+            {'table_name': 'payments', 'table_schema': 'pyexasol_test', 'table_is_virtual': False,
+             'table_has_distribution_key': False, 'table_comment': None},
+            {'table_name': 'users', 'table_schema': 'pyexasol_test', 'table_is_virtual': False,
+             'table_has_distribution_key': False, 'table_comment': None}
+        ]
+        actual = connection.ext.get_sys_tables()
```

Comment:
Again, the order.

Reply:
Thanks for mentioning, ordering is here stable to.